### ISSUE TITLE
Poor results and inconsistent checkpoint file size during training

### ISSUE THREAD

**Issue by RafaKeusch** [SmartForest-no/ForestFormer3D#12]
Dear Binbin Xiang,

Thanks for the great work! I have tested the provided pretrained model and it works out of the box not too bad for my own data. Now I am trying to train the model from scratch with the provided training data but I met the following problems which really confuses me:

1.  The checkpoints I got has a size of 159.3 MB and the pretrained chekpoint `epoch_3000_fix.pth` has a size of 230.2MB, which makes me think I probably did something wrong but cannot find the reason. I tried to use `tools/fix_spconv_checkpoint`, but it did minimal change.

2. The loss curve goes down quite nicely but the val performance was really bad:

<img width="411" height="489" alt="Image" src="https://github.com/user-attachments/assets/04fcf341-613b-4c18-bb22-7d315c11b8ca" />

<img width="411" height="489" alt="Image" src="https://github.com/user-attachments/assets/94719223-bfe1-437d-8719-98a1750d0ac8" />

3. You mentioned you trained the model with a single A100, but I am only using a 4090 24GB, and the VRAM is absolutely not an issue during training, the VRAM usage hovers only around 1-4GB, which again looks very suspicious and makes me think that I did something wrong.

<img width="1090" height="300" alt="Image" src="https://github.com/user-attachments/assets/35748293-e094-40fd-8f9d-407aa7ba1595" />

*One additional notes:
I subsampled the two big bluecat file to accelerate the training:

<img width="1016" height="832" alt="Image" src="https://github.com/user-attachments/assets/f6a6ffc8-1ecf-43c3-8416-d8344e2e417f" />

Could you please provide some hints on what could potentially go wrong in my case?

Thank you very much!

**Comment by bxiang233** [comment]
> Dear Binbin Xiang,
> 
> Thanks for the great work! I have tested the provided pretrained model and it works out of the box not too bad for my own data. Now I am trying to train the model from scratch with the provided training data but I met the following problems which really confuses me:
> 
> 1. The checkpoints I got has a size of 159.3 MB and the pretrained chekpoint `epoch_3000_fix.pth` has a size of 230.2MB, which makes me think I probably did something wrong but cannot find the reason. I tried to use `tools/fix_spconv_checkpoint`, but it did minimal change.
> 2. The loss curve goes down quite nicely but the val performance was really bad:
> 
> <img alt="Image" width="411" height="489" src="https://private-user-images.githubusercontent.com/245014627/516403711-04fcf341-613b-4c18-bb22-7d315c11b8ca.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM2MzAyNjYsIm5iZiI6MTc2MzYyOTk2NiwicGF0aCI6Ii8yNDUwMTQ2MjcvNTE2NDAzNzExLTA0ZmNmMzQxLTYxM2ItNGMxOC1iYjIyLTdkMzE1YzExYjhjYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMFQwOTEyNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNjQ0NGJlMzc4MWMwYThkNTJiNzgxM2U1ODU1Yjc2NTBhMjE2MWEwOTA3ZGQ1ZDY4OTNkYTdkZDkxYzUzZmUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.r0FvRFwrSX42reCWo4S0kj4lQGy2Zfw8-KQTaj9M2Q0"> <img alt="Image" width="411" height="489" src="https://private-user-images.githubusercontent.com/245014627/516403849-94719223-bfe1-437d-8719-98a1750d0ac8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM2MzAyNjYsIm5iZiI6MTc2MzYyOTk2NiwicGF0aCI6Ii8yNDUwMTQ2MjcvNTE2NDAzODQ5LTk0NzE5MjIzLWJmZTEtNDM3ZC04NzE5LTk4YTE3NTBkMGFjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMFQwOTEyNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOGIxZmFjZjhjMDJiM2ZlOTkwMmJiZjY4NmQxMTZmYzg5MzAyZDRiZGNmYzU0YTVmMGJlY2E3ODhjNDViMTNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.B0F1kqQ6TBH-Nx3AO8fH63NmDvK1deF7lSMNCEfnmlw">
> 3. You mentioned you trained the model with a single A100, but I am only using a 4090 24GB, and the VRAM is absolutely not an issue during training, the VRAM usage hovers only around 1-4GB, which again looks very suspicious and makes me think that I did something wrong.
> 
> <img alt="Image" width="1090" height="300" src="https://private-user-images.githubusercontent.com/245014627/516405517-35748293-e094-40fd-8f9d-407aa7ba1595.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM2MzAyNjYsIm5iZiI6MTc2MzYyOTk2NiwicGF0aCI6Ii8yNDUwMTQ2MjcvNTE2NDA1NTE3LTM1NzQ4MjkzLWUwOTQtNDBmZC04ZjlkLTQwN2FhN2JhMTU5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMFQwOTEyNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NTE1NjY2YzY4NWMxZWY3NDhkMDUyMzYxYmFlMGI3OGM4ZGQ3NmRjZjAzMmI5YzI0OTg0YjljOTAyYWY2NGQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.U83Gg5dr2lT0AcT8E7BT3MpT5uOUAoa88Q4aLFXFrpg">
> *One additional notes: I subsampled the two big bluecat file to accelerate the training:
> 
> <img alt="Image" width="1016" height="832" src="https://private-user-images.githubusercontent.com/245014627/516410076-f6a6ffc8-1ecf-43c3-8416-d8344e2e417f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM2MzAyNjYsIm5iZiI6MTc2MzYyOTk2NiwicGF0aCI6Ii8yNDUwMTQ2MjcvNTE2NDEwMDc2LWY2YTZmZmM4LTFlY2YtNDNjMy04NDE2LWQ4MzQ0ZTJlNDE3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMFQwOTEyNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNjFkYzhkODhlMjIwNmZkMThlNTEwZTMyNWJlZjllYjhlYzU0MjVhNzUyNWZiZWU2ZWNiYjRmYTcyMTA2ZWFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZVInkcK8okEGO0SM4oF9lun_HUDld9L3Y5RvipD9Ays">
> Could you please provide some hints on what could potentially go wrong in my case?
> 
> Thank you very much!

Hi, from the information and screenshots you shared, I unfortunately cannot spot a clear mistake in your setup yet. To better understand what is happening, could you please also share the validation curves for: semantic segmentation mIoU, and F1-score over the training epochs? (I usually see F1 to check whether instance training actually works and would like to confirm that the semantic is also working as expected.)

Regarding the low GPU memory usage: this is normal in the early epochs. Our configs train only the backbone first (https://github.com/SmartForest-no/ForestFormer3D/blob/267b648a478a1177bee055ff650a5f4565805e51/configs/oneformer3d_qs_radius16_qp300_2many.py#L23), and once the decoder starts training, memory usage should increase.

Also, could you think through whether you modified anything in the code or configs? Even small changes can lead to the behavior you’re seeing.
